src/adapter/src/coord/timestamp_oracle/catalog_oracle.rs

```diff
@@ -217,9 +217,12 @@ where
 
     async fn read_ts(&self) -> T {
         let ts = self.timestamp_oracle.read_ts();
+        // WIP(joe): confirm!
```

Comment:
The change is from `less_than` to `less_equal`, right? Looking at the catalog oracle, this doesn't seem correct: 

```Rust
    async fn maybe_allocate_new_timestamps(&mut self, ts: &T) {
        if self.durable_timestamp.less_equal(ts)
            // Since the timestamp is at its max value, we know that no other Coord can
            // allocate a higher value.
            && self.durable_timestamp.less_than(&T::maximum())
        {
            self.durable_timestamp = ts.step_forward_by(&self.persist_interval);
            let res = self
                .timestamp_persistence
                .persist_timestamp(self.durable_timestamp.clone())
                .await;

            res.unwrap_or_terminate("can't persist timestamp");
        }
    }
```

Specifically: `if self.durable_timestamp.less_equal(ts)`.

When a new oracle starts up, I it uses the persisted timestamp exactly without incrementing it. Meaning that the persisted timestamp must always be higher than served timestamps.

Which tests required this change?

Reply:
Ah right, that makes sense. The `Timestamp::MAX` scenario really throws a wrench into a lot of things. I think this is correct then.